### PR TITLE
Adds support for s3 folders.

### PR DESCRIPTION
**Pull request by bantonj**

### CODE REVIEW

File: pushd_plugin.go [hunk 1]
```diff
@@ -48,14 +51,28 @@ func fileNameToParams(requestUri string) string {
 
 }
 
+// removes pushdPath and uuid from path if they exist
```
Comment (jtomson) [inline pushd_plugin.go:54]: don't we have per-user-id prefixes in s3? I'm not clear why we remove them

**Review by jtomson — COMMENTED** [review]
can you add an example merci!

**Comment by bantonj** [comment]
This change supports using user-id prefixes or not for avatars. So you can use:
https://staging-imgproxy.pushd.com/2bf38af4-4714-11eb-bde6-0a3dff0b2a67/avatars/0001140C-5685-4334-AA60-EA251EEC386E.jpeg
or
https://staging-imgproxy.pushd.com/avatars/0001140C-5685-4334-AA60-EA251EEC386E.jpeg

It removes the user_id uuid in getS3Path because it's getting the s3 path for downloading the original. I'll make that clearer.

**Comment by bantonj** [comment]
Examples:
https://example.com/s3-folder/example_file.jpeg
https://example.com/uuid/s3-folder/example_file.jpeg

**Review by jtomson — APPROVED** [review]
tres bien!

**Pull request merged**